syntax = "proto3";


message AuthMessage {
    oneof payload {
        MsgInit      init = 1;
        MsgChallenge challenge = 2;
        MsgResponse  response = 3;
        MsgResult    result = 4;
    }
}

message MsgInit {
    int32 serial_id = 1;
    bytes signature = 2;
}

message MsgChallenge {
    bytes random_data = 1;
}

message MsgResponse {
    bytes signature = 1;
}

message MsgResult {
    bool success = 1;
    string message = 2;
}